File: app/src/screens/chatscreen.js
import { Send, ArrowLeft, WifiOff } from 'lucide-react';
import { useState } from 'react';

interface ChatScreenProps {
  onBack: () => void;
}

interface Message {
  text: string;
  isUser: boolean;
}

export function ChatScreen({ onBack }: ChatScreenProps) {
  const [messages, setMessages] = useState<Message[]>([
    { text: 'Hello! I\'m your offline expert assistant. How can I help you today?', isUser: false }
  ]);
  const [inputText, setInputText] = useState('');

  const suggestions = [
    'Give me a tip',
    'Explain this',
    'Best practices',
    'How to improve'
  ];

  const handleSend = () => {
    if (!inputText.trim()) return;

    setMessages([...messages,
      { text: inputText, isUser: true },
      { text: 'I understand your question. Here\'s my expert advice based on my offline knowledge...', isUser: false }
    ]);
    setInputText('');
  };

  const handleSuggestion = (suggestion: string) => {
    setMessages([...messages,
      { text: suggestion, isUser: true },
      { text: 'Great question! Let me provide you with expert guidance on this topic...', isUser: false }
    ]);
  };

  return (
    <div className="min-h-screen bg-gray-50 dark:bg-gray-900 flex flex-col">
      <div className="bg-gradient-to-r from-green-400 to-emerald-500 dark:from-green-600 dark:to-emerald-700 text-white px-6 py-4 flex items-center gap-4">
        <button
          onClick={onBack}
          className="w-10 h-10 bg-white/20 rounded-full flex items-center justify-center hover:bg-white/30 transition-colors"
        >
          <ArrowLeft className="w-5 h-5" />
        </button>
        <div className="flex-1">
          <h1 className="text-white">Expert Chat</h1>
          <div className="flex items-center gap-1 text-sm opacity-90">
            <WifiOff className="w-3 h-3" />
            <span>Offline</span>
          </div>
        </div>
      </div>

      <div className="flex-1 overflow-y-auto px-6 py-6 space-y-4">
        {messages.map((message, index) => (
          <div key={index} className={`flex ${message.isUser ? 'justify-end' : 'justify-start'}`}>
            <div className={`max-w-[80%] p-4 rounded-2xl ${
              message.isUser
                ? 'bg-gradient-to-br from-green-400 to-emerald-500 dark:from-green-600 dark:to-emerald-700 text-white'
                : 'bg-white dark:bg-gray-800 text-gray-900 dark:text-white shadow-sm'
            }`}>
              <p className="text-sm leading-relaxed">{message.text}</p>
            </div>
          </div>
        ))}

        {messages.length <= 1 && (
          <div className="space-y-3 mt-6">
            <p className="text-gray-500 dark:text-gray-400 text-sm text-center mb-4">Quick suggestions:</p>
            <div className="grid grid-cols-2 gap-2">
              {suggestions.map((suggestion, index) => (
                <button
                  key={index}
                  onClick={() => handleSuggestion(suggestion)}
                  className="bg-white dark:bg-gray-800 text-gray-700 dark:text-gray-300 px-4 py-3 rounded-xl text-sm hover:shadow-md transition-shadow border border-gray-200 dark:border-gray-700"
                >
                  {suggestion}
                </button>
              ))}
            </div>
          </div>
        )}
      </div>

      <div className="p-4 bg-white dark:bg-gray-800 border-t border-gray-200 dark:border-gray-700">
        <div className="flex gap-2">
          <input
            type="text"
            value={inputText}
            onChange={(e) => setInputText(e.target.value)}
            onKeyPress={(e) => e.key === 'Enter' && handleSend()}
            placeholder="Type your question..."
            className="flex-1 bg-gray-100 dark:bg-gray-700 text-gray-900 dark:text-white px-4 py-3 rounded-xl focus:outline-none focus:ring-2 focus:ring-green-400 dark:focus:ring-green-600 placeholder-gray-500 dark:placeholder-gray-400"
          />
          <button
            onClick={handleSend}
            className="bg-gradient-to-br from-green-400 to-emerald-500 dark:from-green-600 dark:to-emerald-700 text-white w-12 h-12 rounded-xl flex items-center justify-center hover:shadow-lg transition-shadow"
          >
            <Send className="w-5 h-5" />
          </button>
        </div>
      </div>
    </div>
  );
}

File: app/src/main/java/com/runanywhere/startup_hackathon20/ui_screens/chatscreen.kt
package com.runanywhere.startup_hackathon20.ui_screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.lazy.rememberLazyListState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.TextFieldValue
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.runanywhere.startup_hackathon20.ChatViewModel
import com.runanywhere.startup_hackathon20.ChatMessage
import com.runanywhere.startup_hackathon20.R
import com.runanywhere.startup_hackathon20.ui.theme.Startup_hackathon20Theme
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.launch

data class Message(
    val text: String,
    val isUser: Boolean
)

@Composable
fun ChatScreen(
    onBack: () -> Unit,
    viewModel: ChatViewModel? = viewModel()
) {
    var inputText by remember { mutableStateOf(TextFieldValue("")) }

    // Collect messages from ViewModel
    val messages by (viewModel?.messages ?: MutableStateFlow(emptyList())).collectAsState()
    val isLoading by (viewModel?.isLoading ?: MutableStateFlow(false)).collectAsState()
    val currentModelId by (viewModel?.currentModelId
        ?: MutableStateFlow<String?>(null)).collectAsState()
    val statusMessage by (viewModel?.statusMessage ?: MutableStateFlow("Initializing...")).collectAsState()

    val listState = rememberLazyListState()
    val coroutineScope = rememberCoroutineScope()

    // Auto-scroll to bottom when new messages arrive
    LaunchedEffect(messages.size) {
        if (messages.isNotEmpty()) {
            coroutineScope.launch {
                listState.animateScrollToItem(messages.size - 1)
            }
        }
    }

    val suggestions = listOf(
        "Give me a tip",
        "Explain this",
        "Best practices",
        "How to improve"
    )

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color(0xFFF7F7F7))
    ) {

        // 🔥 TOP BAR WITH GRADIENT
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(
                    Brush.horizontalGradient(
                        listOf(
                            Color(0xFF4CAF50),
                            Color(0xFF2ecc71)
                        )
                    )
                )
                .padding(16.dp)
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {

                // BACK BUTTON
                IconButton(
                    onClick = onBack,
                    modifier = Modifier
                        .size(48.dp)
                        .clip(CircleShape)
                        .background(Color.White.copy(alpha = 0.2f))
                ) {
                    Icon(
                        painter = painterResource(id = R.drawable.ic_arrow_left),
                        contentDescription = "Back",
                        tint = Color.White
                    )
                }

                Spacer(Modifier.width(12.dp))

                Column(modifier = Modifier.weight(1f)) {
                    Text(
                        "Expert Chat",
                        color = Color.White,
                        fontWeight = FontWeight.Bold
                    )
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            painter = painterResource(
                                id = if (currentModelId != null) R.drawable.ic_wifi_off else R.drawable.ic_wifi_off
                            ),
                            contentDescription = "",
                            tint = Color.White,
                            modifier = Modifier.size(14.dp)
                        )
                        Spacer(Modifier.width(4.dp))
                        Text(
                            if (currentModelId != null) "Model Loaded" else "No Model",
                            color = Color.White.copy(0.9f),
                            style = MaterialTheme.typography.bodySmall
                        )
                    }
                }

                // Clear Chat Button
                if (messages.isNotEmpty()) {
                    var showDialog by remember { mutableStateOf(false) }

                    IconButton(
                        onClick = { showDialog = true },
                        modifier = Modifier
                            .size(40.dp)
                            .clip(CircleShape)
                            .background(Color.White.copy(alpha = 0.2f))
                    ) {
                        Icon(
                            painter = painterResource(id = R.drawable.ic_delete),
                            contentDescription = "Clear Chat",
                            tint = Color.White,
                            modifier = Modifier.size(20.dp)
                        )
                    }

                    if (showDialog) {
                        AlertDialog(
                            onDismissRequest = { showDialog = false },
                            title = { Text("Clear Chat History") },
                            text = { Text("Are you sure you want to delete all chat messages? This action cannot be undone.") },
                            confirmButton = {
                                TextButton(
                                    onClick = {
                                        viewModel?.clearAllMessages()
                                        showDialog = false
                                    }
                                ) {
                                    Text("Clear")
                                }
                            },
                            dismissButton = {
                                TextButton(onClick = { showDialog = false }) {
                                    Text("Cancel")
                                }
                            }
                        )
                    }
                }
            }
        }

        // 🔥 MESSAGES LIST
        LazyColumn(
            state = listState,
            modifier = Modifier
                .weight(1f)
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(10.dp)
        ) {
            // Show welcome message if no messages
            if (messages.isEmpty()) {
                item {
                    Box(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(vertical = 20.dp),
                        contentAlignment = Alignment.Center
                    ) {
                        Column(horizontalAlignment = Alignment.CenterHorizontally) {
                            Text(
                                "Hello! I'm your offline expert assistant.",
                                style = MaterialTheme.typography.bodyLarge,
                                color = Color.Gray
                            )
                            Spacer(Modifier.height(8.dp))
                            Text(
                                statusMessage,
                                style = MaterialTheme.typography.bodyMedium,
                                color = Color.Gray
                            )
                        }
                    }
                }
            }

            items(messages) { message ->
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = if (message.isUser) Arrangement.End else Arrangement.Start
                ) {

                    Box(
                        modifier = Modifier
                            .then(
                                if (message.isUser) {
                                    Modifier.background(
                                        brush = Brush.linearGradient(
                                            listOf(
                                                Color(0xFF4CAF50),
                                                Color(0xFF2ECC71)
                                            )
                                        ),
                                        shape = RoundedCornerShape(20.dp)
                                    )
                                } else {
                                    Modifier.background(
                                        color = Color.White,
                                        shape = RoundedCornerShape(20.dp)
                                    )
                                }
                            )
                            .padding(14.dp)
                            .widthIn(max = 260.dp)
                    ) {
                        Text(
                            message.text,
                            color = if (message.isUser) Color.White else Color.Black
                        )
                    }
                }
            }

            // 🔥 SUGGESTIONS when no messages
            if (messages.isEmpty()) {
                item {
                    Spacer(Modifier.height(20.dp))
                    Text(
                        "Quick suggestions:",
                        color = Color.Gray,
                        style = MaterialTheme.typography.bodyMedium
                    )
                    Spacer(Modifier.height(10.dp))

                    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                        suggestions.chunked(2).forEach { rowItems ->
                            Row(
                                Modifier.fillMaxWidth(),
                                horizontalArrangement = Arrangement.spacedBy(8.dp)
                            ) {
                                rowItems.forEach { suggestion ->
                                    Button(
                                        onClick = {
                                            viewModel?.sendMessage(suggestion)
                                        },
                                        modifier = Modifier.weight(1f),
                                        colors = ButtonDefaults.buttonColors(
                                            containerColor = Color.White
                                        ),
                                        enabled = currentModelId != null && !isLoading
                                    ) {
                                        Text(suggestion, color = Color.Black)
                                    }
                                }
                            }
                        }
                    }
                }
            }

            // Show loading indicator
            if (isLoading) {
                item {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.Start
                    ) {
                        Box(
                            modifier = Modifier
                                .background(
                                    color = Color.White,
                                    shape = RoundedCornerShape(20.dp)
                                )
                                .padding(14.dp)
                        ) {
                            CircularProgressIndicator(
                                modifier = Modifier.size(20.dp),
                                color = Color(0xFF4CAF50),
                                strokeWidth = 2.dp
                            )
                        }
                    }
                }
            }
        }

        // 🔥 INPUT FIELD + SEND BUTTON
        Row(
            modifier = Modifier
                .background(Color.White)
                .padding(10.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {

            OutlinedTextField(
                value = inputText,
                onValueChange = { inputText = it },
                placeholder = { Text("Type your question...") },
                modifier = Modifier
                    .weight(1f)
                    .border(1.dp, Color.LightGray, RoundedCornerShape(12.dp)),
                shape = RoundedCornerShape(12.dp)
            )

            Spacer(Modifier.width(10.dp))

            IconButton(
                onClick = {
                    if (inputText.text.isNotBlank()) {
                        viewModel?.sendMessage(inputText.text)
                        inputText = TextFieldValue("")
                    }
                },
                enabled = currentModelId != null && !isLoading && inputText.text.isNotBlank(),
                modifier = Modifier
                    .size(50.dp)
                    .clip(RoundedCornerShape(12.dp))
                    .background(
                        if (currentModelId != null && !isLoading && inputText.text.isNotBlank()) {
                            Brush.linearGradient(
                                listOf(
                                    Color(0xFF4CAF50),
                                    Color(0xFF2ECC71)
                                )
                            )
                        } else {
                            Brush.linearGradient(
                                listOf(
                                    Color.Gray.copy(alpha = 0.5f),
                                    Color.Gray.copy(alpha = 0.5f)
                                )
                            )
                        }
                    )
            ) {
                if (isLoading) {
                    CircularProgressIndicator(
                        modifier = Modifier.size(24.dp),
                        color = Color.White,
                        strokeWidth = 2.dp
                    )
                } else {
                    Icon(
                        painter = painterResource(id = R.drawable.ic_send),
                        contentDescription = "Send",
                        tint = Color.White
                    )
                }
            }
        }
    }
}

@Preview(showBackground = true, showSystemUi = true)
@Composable
fun ChatScreenPreview() {
    Startup_hackathon20Theme {
        ChatScreen(onBack = {}, viewModel = null)
    }
}
